controllers/users.js

```diff
@@ -34,9 +34,33 @@ function update(req, res, next) {
     .catch((err) => next(err))
 }
 
+function setProfile(req, res, next) {
+  const {
+    name, email, password, geolocation,
+  } = req.body
+
+  User.findByIdAndUpdate(req.user.id, {
```

Comment:
we have user instance so no need to find here. Try to using req.user.set then req.user.save